src/modules/map/components/cesium/utils/addImageryLayer-mixins.js

```diff
@@ -57,18 +42,12 @@ const addImageryLayerMixins = {
             this.layer = this.createImagery(newUrl)
             viewer.scene.imageryLayers.add(this.layer, index)
         },
-        zIndex(zIndex) {
+        zIndex() {
             if (this.layer) {
                 const imageryLayers = this.getViewer().scene.imageryLayers
-                const index = imageryLayers.indexOf(this.layer)
-                const indexDiff = Math.abs(zIndex - index)
-                for (let i = indexDiff; i !== 0; i--) {
-                    if (index > zIndex) {
-                        imageryLayers.lower(this.layer)
-                    } else {
-                        imageryLayers.raise(this.layer)
-                    }
-                }
+                imageryLayers.lowerToBottom(this.layer)
+                // raise one time to place layer above base layer
```

Comment:
somehow this works (when re-ordering layers), I don't really get why?

Reply:
Indeed, it was working wrong when more than 2 layers. I updated this function